src/ert/_c_wrappers/job_queue/queue_differ.py

```diff
@@ -52,7 +52,7 @@ def snapshot(self) -> Dict[int, str]:
                 snapshot[self._qindex_to_iens[q_index]] = st
             except KeyError as e:
                 logger.debug(f"differ could produce no snapshot due to {e}")
-                return None
+                return snapshot
```

Comment:
This looks like it is now possible to return a partial snapshot? Intentional? 

If `None` should no longer be returned, the docstring should be updated.

Reply:
Ah yes, I forgot about this case. Indeed this would make a partial snapshot be returned in case of a failure. I think the callsite does not consider that None could be returned in which case you get an exception. Should we reraise the KeyError after logging?

Reply:
`_queue_changes` similarly indescriminately calls `items` on it as well.

Reply:
Reraising KeyError is probably the way to go yes.

Reply:
Added a separate commit for the reraise